server/util/tokenBalanceCache.ts

```diff
@@ -92,6 +101,56 @@ export class TokenBalanceProvider {
     return new BN(balanceBigNum.toString());
   }
 
+  public async getCosmosTokenBalance(url: string, network: string, userAddress: string): Promise<BN> {
+    /* also do network === ChainNetwork.NativeCosmos / Terra or ChainNetwork.CosmosNFT => should check NFTs */ 
+    
+    let balance;
+    let api;
+    if (network === ChainNetwork.Terra) {
+
+      api = new LCDClient({
+        URL: url + `/node_info?key=${process.env.TERRA_SETTEN_PHOENIX_API_KEY}`, 
+        chainID: 'phoenix-1',
+      });
+
+      try {
+        /* terra.js staking.parameter call returns 
+           Cannot read property 'unbonding_time' of undefined unbonding_time */
+
+        /* bank/v1beta1/denoms_metadata/uluna => exponent => look for luna???*/
+        const bal = await api.bank.balance(userAddress);
+        (bal[0].get('uluna')) ? balance = new BN(bal[0].get('uluna').toString()) : 
```

Comment:
Should rewrite this as an if/else (or if maintaining conciseness, using an `||` operator) rather than a ternary.